module/VuFind/src/VuFind/Db/Service/LoginTokenService.php

```diff
@@ -108,7 +106,37 @@ public function createAndPersistToken(
      */
     public function matchToken(array $token): ?LoginTokenEntityInterface
     {
-        return $this->getDbTable('LoginToken')->matchToken($token);
+        $userId = null;
+        foreach ($this->getBySeries($token['series']) as $row) {
```

Comment:
getBySeries() is supposed to return an array of LoginTokenEntityInterface objects, but the `$row` object here seems to be getting treated as an old-fashioned Laminas row object (since it's being interchangeably treated as an object or an array). I think this needs to be revised to use the getters from the interface instead of direct data access.